lib/Sema/CSApply.cpp

```diff
@@ -424,7 +422,7 @@ namespace {
                 // FIXME: the hop through 'getDecl()' is because
                 // SpecializedProtocolConformance doesn't substitute into
                 // witnesses' ConcreteDeclRefs.
-                Type expectedFnType = simplifiedFnType->getResult();
+                Type expectedFnType = simplifyType(overload.openedType);
```

Comment:
I think you need the full opened type here.

Reply:
@CodaFi Hmm, shouldn't the result of the full opened type be the opened type in this case?

Reply:
Let’s add an assertion to that effect.

Reply:
@CodaFi Sure thing! How does the updated code look?

Reply:
Great, thanks!